src/State/Provider/ReadProvider.php

```diff
@@ -84,6 +88,8 @@ public function provide(Operation $operation, array $uriVariables = [], array $c
                 || ($operation instanceof Put && !($operation->getAllowCreate() ?? false))
             )
         ) {
+            // In case the dev just forgot to implement it
+            $this->logger?->debug('No provider registered for {resource_class}', ['resource_class' => $resourceClass]);
             throw new NotFoundHttpException('Not Found');
```

Comment:
we could also keep the previous exception (ProviderNotFound) and use it here as `previous` argument of `NotFoundHttpException`

Reply:
That's a clear benefit of moving it to line 81. I'll do that.